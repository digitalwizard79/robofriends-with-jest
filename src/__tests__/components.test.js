import { shallow } from 'enzyme'
import React from 'react'
import CounterButton from '../components/CounterButton'
import CardList from '../components/CardList'
import Card from '../components/Card'
import MainPage from '../components/MainPage'

describe('MainPage.js', () => {
    let wrapper;
    beforeEach(() => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            robots: [],
            searchField: '',
            isPending: false
        }
        wrapper = shallow(<MainPage { ...mockProps } />)
    })

    it('renders MainPage without crashing', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Filters robots correctly', () => {
        const mockProps2 = {
            onRequestRobots: jest.fn(),
            robots: [{
                id: 3,
                name: 'John',
                email: 'john@gmail.com'
            }],
            searchField: 'john',
            isPending: false
        }

        let wrapper2 = shallow(<MainPage { ...mockProps2 } />)

        //expect(wrapper2.instance().filterRobots([])).toEqual([])
        expect(wrapper2.instance().filterRobots()).toEqual([{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }])


        const mockProps3 = {
            onRequestRobots: jest.fn(),
            robots: [{
                id: 3,
                name: 'John',
                email: 'john@gmail.com'
            }],
            searchField: 'a',
            isPending: false
        }

        const filteredRobots = []
        let wrapper3 = shallow(<MainPage { ...mockProps3 } />)        
        expect(wrapper3.instance().filterRobots()).toEqual([])
    })

    it('Check isPending', () => {
        const mockProps4 = {
            onRequestRobots: jest.fn(),
            robots: [],
            searchField: '',
            isPending: true
        }
        let wrapper4 = shallow(<MainPage { ...mockProps4 } />)
        expect(mockProps4.isPending).toEqual(true)
    })
})

describe('Card', () => {
    it('Expect to render Card component', () => {    
        expect(shallow(<Card name='Test' email='abc@some.com' />)).toMatchSnapshot()
    })
})

describe('CardList', () => {
    it('Expect to render CardList component', () => {    
        const mockRobots = [
            {
                id: 1,
                name: 'John Snow',
                username: 'JohnJohn',
                email: 'john@gmail.com'
            }
        ]
        expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot()
    })
})

describe('CounterButton', () => {
    it('Expect to render CounterButton component', () => {    
        const mockColor = 'red'
        const wrapper = shallow(<CounterButton color={mockColor} />)
    
        expect(wrapper).toMatchSnapshot()
    })
    
    it('correctly increments the counter', () => {
        const mockColor = 'red'
        const wrapper = shallow(<CounterButton color={mockColor} />)
    
        wrapper.find('[id="counter"]').simulate('click')
        expect(wrapper.state()).toEqual({ count: 1 })
    })
    
    it('check for shouldComponentUpdate', () => {
        const mockCB = new CounterButton()    
        expect(mockCB.shouldComponentUpdate({}, { count: 0 })).toEqual(false)
        expect(mockCB.shouldComponentUpdate({}, { count: 1 })).toEqual(true)
    })
})