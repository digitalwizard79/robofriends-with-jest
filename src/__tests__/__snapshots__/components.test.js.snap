// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Expect to render Card component 1`] = `
<div
  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?size=200x200"
  />
  <div>
    <h2>
      Test
    </h2>
    <p>
      abc@some.com
    </p>
  </div>
</div>
`;

exports[`CardList Expect to render CardList component 1`] = `
<div>
  <Card
    email="john@gmail.com"
    id={1}
    key="0"
    name="John Snow"
  />
</div>
`;

exports[`CounterButton Expect to render CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;

exports[`MainPage.js renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={[Function]}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
